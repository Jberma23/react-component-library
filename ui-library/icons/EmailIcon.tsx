export const EmailIcon = () => (
  <svg
    width="14"
    height="12"
    viewBox="0 0 14 12"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1.36152 0.51001C0.611523 0.51001 0.0615234 1.11001 0.0615234 1.86001V9.96001C0.0615234 10.71 0.661523 11.31 1.36152 11.31H12.6115C13.3615 11.31 13.9115 10.71 13.9115 9.96001V1.86001C13.9115 1.11001 13.3115 0.51001 12.6115 0.51001H1.36152ZM12.3615 3.36001L6.96152 6.66001L1.56152 3.36001V2.06001L6.96152 5.36001L12.3615 2.06001C12.3615 2.06001 12.3615 3.36001 12.3615 3.36001Z"
      fill="black"
    />
  </svg>
);
