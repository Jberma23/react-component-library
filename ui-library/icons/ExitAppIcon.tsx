export const ExitAppIcon = (props: JSX.IntrinsicElements["svg"]) => (
  <svg
    width="16"
    height="16"
    viewBox="0 0 16 16"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M7.33333 3.33331V4.66665H3.33333V12.6666H11.3333V8.66665H12.6667V14H2V3.33331H7.33333ZM14.6667 1.33331V6.66665H13.3333V3.60931L7.80474 9.13805L6.86193 8.19524L12.3893 2.66665H9.33333V1.33331H14.6667Z"
      fill="#4657CE"
    />
  </svg>
);
