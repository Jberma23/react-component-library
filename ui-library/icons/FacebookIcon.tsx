export const FacebookIcon = () => (
  <svg
    width="8"
    height="15"
    viewBox="0 0 8 15"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M7.59351 0.945496H5.79351C3.74351 0.945496 2.39351 2.2955 2.39351 4.3455V5.9455H0.593506C0.443506 5.9455 0.343506 6.0955 0.343506 6.1955V8.4955C0.343506 8.6455 0.493506 8.7455 0.593506 8.7455H2.44351V14.5455C2.44351 14.6955 2.59351 14.7955 2.69351 14.7955H5.09351C5.24351 14.7955 5.34351 14.6455 5.34351 14.5455V8.79549H7.49351C7.64351 8.79549 7.74351 8.64549 7.74351 8.54549V6.2455C7.74351 6.0955 7.59351 5.9955 7.49351 5.9955H5.39351V4.6455C5.39351 3.9955 5.54351 3.6455 6.39351 3.6455H7.59351C7.74351 3.6455 7.84351 3.4955 7.84351 3.3955V1.1955C7.84351 1.0455 7.74351 0.945496 7.59351 0.945496Z"
      fill="black"
    />
  </svg>
);
