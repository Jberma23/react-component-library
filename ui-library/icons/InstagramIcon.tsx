export const InstagramIcon = () => (
  <svg
    width="16"
    height="16"
    viewBox="0 0 16 16"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7.99277 1.94448C9.99277 1.94448 10.1928 1.94448 10.9428 1.99448C11.3928 1.99448 11.8928 2.09448 12.2928 2.24448C12.9428 2.44448 13.4428 2.99448 13.6928 3.64448C13.8928 4.09448 13.9428 4.54448 13.9428 4.99448C13.9928 5.74448 13.9928 5.99448 13.9928 7.94448C13.9928 9.89448 13.9928 10.1445 13.9428 10.8945C13.9428 11.3445 13.8428 11.8445 13.6928 12.2445C13.4428 12.8945 12.9428 13.3945 12.2928 13.6445C11.8428 13.8445 11.3928 13.8945 10.9428 13.8945C10.1928 13.9445 9.94277 13.9445 7.99277 13.9445C6.04277 13.9445 5.79277 13.9445 5.04277 13.8945C4.59277 13.8945 4.09277 13.7945 3.69277 13.6445C3.34277 13.5445 3.04277 13.3945 2.79277 13.1445C2.54277 12.9445 2.34277 12.5945 2.24277 12.2945C2.04277 11.8445 1.99277 11.3945 1.99277 10.9445C1.99277 10.1445 1.99277 9.89448 1.99277 7.94448C1.99277 5.99448 1.99277 5.74448 2.04277 4.99448C2.04277 4.54448 2.14277 4.04448 2.29277 3.64448C2.39277 3.29448 2.54277 3.04448 2.79277 2.79448C2.99277 2.54448 3.29277 2.34448 3.64277 2.24448C4.09277 2.04448 4.54277 1.99448 4.99277 1.99448C5.79277 1.94448 5.99277 1.94448 7.99277 1.94448ZM7.99277 0.594482C5.94277 0.594482 5.74277 0.594482 4.94277 0.644482C4.29277 0.644482 3.74277 0.794482 3.14277 1.04448C2.69277 1.19448 2.24277 1.49448 1.89277 1.89448C1.49277 2.24448 1.19277 2.69448 1.04277 3.19448C0.792773 3.74448 0.692773 4.34448 0.642773 4.99448C0.592773 5.74448 0.592773 5.99448 0.592773 7.99448C0.592773 9.99448 0.592773 10.2445 0.642773 11.0445C0.642773 11.6945 0.792773 12.2445 0.992773 12.8445C1.19277 13.3445 1.49277 13.7945 1.84277 14.1445C2.19277 14.4945 2.64277 14.8445 3.14277 14.9945C3.74277 15.2445 4.29277 15.2945 4.94277 15.3445C5.74277 15.3945 5.99277 15.3945 7.99277 15.3945C9.99277 15.3945 10.2428 15.3945 11.0428 15.3445C11.6928 15.3445 12.2428 15.1945 12.8428 14.9945C13.3428 14.7945 13.7928 14.4945 14.1428 14.1445C14.4928 13.7945 14.8428 13.3445 14.9928 12.8445C15.1928 12.2445 15.2928 11.6945 15.3428 11.0445C15.3928 10.2445 15.3928 9.99448 15.3928 7.99448C15.3928 5.99448 15.3928 5.74448 15.3428 4.94448C15.3428 4.29448 15.1928 3.74448 14.9928 3.14448C14.7928 2.64448 14.4928 2.19448 14.1428 1.84448C13.7928 1.49448 13.3428 1.14448 12.8428 0.994482C12.2428 0.744482 11.6928 0.694482 11.0428 0.644482C10.2428 0.594482 10.0428 0.594482 7.99277 0.594482Z"
      fill="black"
    />
    <path
      d="M7.99238 4.19449C5.94238 4.19449 4.19238 5.89449 4.19238 7.99449C4.19238 10.0945 5.89238 11.7945 7.99238 11.7945C10.0924 11.7945 11.7924 10.0945 11.7924 7.99449C11.7924 5.94449 10.0924 4.19449 7.99238 4.19449ZM7.99238 10.4945C6.64238 10.4945 5.54238 9.39449 5.54238 7.99449C5.54238 6.64449 6.64238 5.54449 7.99238 5.54449C9.34238 5.54449 10.4424 6.64449 10.4424 7.99449C10.4924 9.34449 9.34238 10.4945 7.99238 10.4945Z"
      fill="black"
    />
    <path
      d="M11.943 4.94447C12.44 4.94447 12.843 4.54153 12.843 4.04447C12.843 3.54741 12.44 3.14447 11.943 3.14447C11.4459 3.14447 11.043 3.54741 11.043 4.04447C11.043 4.54153 11.4459 4.94447 11.943 4.94447Z"
      fill="black"
    />
  </svg>
);
