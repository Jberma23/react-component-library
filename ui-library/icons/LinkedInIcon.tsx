export const LinkedInIcon = () => (
  <svg
    width="15"
    height="15"
    viewBox="0 0 15 15"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13.85 0.950012H1.94995C1.39995 0.950012 0.949951 1.40001 0.949951 1.95001V13.9C0.949951 14.45 1.39995 14.9 1.94995 14.9H13.9C14.45 14.9 14.9 14.45 14.9 13.9V1.95001C15 1.40001 14.4 0.950012 13.85 0.950012ZM5.09995 12.85H2.89995V6.10001H4.99995V12.85H5.09995ZM3.99995 5.25001C3.29995 5.25001 2.79995 4.70001 2.79995 4.05001C2.79995 3.40001 3.34995 2.85001 3.99995 2.85001C4.64995 2.80001 5.14995 3.30001 5.19995 3.95001C5.19995 4.00001 5.19995 4.05001 5.19995 4.10001C5.24995 4.75001 4.74995 5.25001 4.09995 5.30001C4.09995 5.25001 4.04995 5.25001 3.99995 5.25001ZM12.85 12.85H10.75V9.20001C10.75 8.10001 10.3 7.75001 9.64995 7.75001C8.99995 7.75001 8.44995 8.20001 8.44995 9.20001V12.85H6.34995V6.10001H8.34995V7.00001C8.74995 6.30001 9.49995 5.90001 10.35 5.90001C11.55 5.90001 12.8 6.55001 12.8 8.65001V12.85H12.85Z"
      fill="black"
    />
  </svg>
);
