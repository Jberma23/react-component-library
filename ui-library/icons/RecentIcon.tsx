export const RecentIcon = (props: JSX.IntrinsicElements["svg"]) => (
  <svg
    width="20"
    height="20"
    viewBox="0 0 23 23"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      d="M10 4V10H14.5M1.68508 13.4442C2.13738 14.5361 2.80031 15.5282 3.63604 16.364C4.47177 17.1997 5.46392 17.8626 6.55585 18.3149C7.64778 18.7672 8.8181 19 10 19C11.1819 19 12.3522 18.7672 13.4442 18.3149C14.5361 17.8626 15.5282 17.1997 16.364 16.364C17.1997 15.5282 17.8626 14.5361 18.3149 13.4442C18.7672 12.3522 19 11.1819 19 10C19 7.61305 18.0518 5.32387 16.364 3.63604C14.6761 1.94821 12.3869 1 10 1C7.61305 1 5.32387 1.94821 3.63604 3.63604C2.66715 4.60493 1.44198 5.73004 1 7M1 7H5.5M1 7V2"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);
