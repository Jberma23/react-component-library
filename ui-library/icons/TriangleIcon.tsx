type TriangleIconProps = JSX.IntrinsicElements["svg"];

export const TriangleIcon = ({ className, ...rest }: TriangleIconProps) => (
  <svg
    className={className}
    height="12"
    width="13"
    viewBox="0 0 13 12"
    xmlns="http://www.w3.org/2000/svg"
    {...rest}
  >
    <path d="M1 11.5V1.20711L11.2929 11.5H1Z" />
  </svg>
);
