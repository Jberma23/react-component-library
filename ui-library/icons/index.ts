// istanbul ignore file

/** Explicitly re-export icons as we need */
export {
  ArrowDownTrayIcon,
  ArrowLeftIcon,
  ArrowRightIcon,
  BoltIcon,
  CalendarIcon,
  ExclamationTriangleIcon,
  UserIcon,
  CheckCircleIcon,
} from "@heroicons/react/24/solid";

export {
  CheckIcon,
  ChevronDownIcon,
  ChevronUpIcon,
  LockClosedIcon,
  LockOpenIcon,
  XMarkIcon,
} from "@heroicons/react/20/solid";

export {
  ArrowDownTrayIcon as ArrowDownTrayOutlineIcon,
  ArrowUturnRightIcon,
  ArrowPathIcon,
  ArrowRightCircleIcon,
  ArrowRightOnRectangleIcon,
  Bars3Icon,
  BellIcon,
  BookmarkIcon,
  CheckCircleIcon as CheckCircleOutlineIcon,
  Cog6ToothIcon,
  MagnifyingGlassIcon,
  MinusIcon,
  PlusIcon,
  UserCircleIcon,
} from "@heroicons/react/24/outline";

export { EmailIcon } from "./EmailIcon";
export { ExitAppIcon } from "./ExitAppIcon";
export { FacebookIcon } from "./FacebookIcon";
export { InstagramIcon } from "./InstagramIcon";
export { LinkedInIcon } from "./LinkedInIcon";
export { TriangleIcon } from "./TriangleIcon";
export { TwitterIcon } from "./TwitterIcon";

export { RecentIcon } from "./RecentIcon";
export { UpdatingIndicatorIcon } from "./UpdatingIndicatorIcon";
