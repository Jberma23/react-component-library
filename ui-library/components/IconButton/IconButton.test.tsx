import { render } from "@testing-library/react";
import { MagnifyingGlassIcon } from "@ui-library/icons";
import { IconButton } from ".";

describe("IconButton", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<IconButton SvgIcon={MagnifyingGlassIcon} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
