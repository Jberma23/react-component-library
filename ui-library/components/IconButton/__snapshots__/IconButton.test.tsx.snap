// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="h-[40px] h-[40px] px-xs border-1 border-transparent rounded-md ring-offset-1 focus-visible:ring-2 focus-visible:ring-teal-500 transition-colors duration-default bg-gray-200 hover:bg-gray-300 active:bg-gray-400 text-black w-fit leading-body font-semibold text-body-regular"
  >
    <div
      class="flex flex-row gap-xxs"
    >
      <div
        class="flex-1"
      >
        <svg
          aria-hidden="true"
          class="fill-current inline"
          data-slot="icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
  </button>
</DocumentFragment>
`;
